### PR TITLE
[DBEX-70682] validate Reserves/Guard dates using DateUI Schema 

### PR DESCRIPTION
**Pull request by freeheeling**
## Summary

It is currently possible to create an "XX" value for the Day field of some dates in the 526 form causing primary and backup path failures after submission due to data invalidation errors. Issues with all date fields determined to have this issue are slotted to be resolved, beginning with this initially identified instance. Where required, date field days in which a user could previously input a value are being updated to dropdown select options based on the selected Month, ensuring reliable data inputs. The DateUI schema was imported directly into and accessed within the impacted page, instead of attempting to assign all UI attributes within the page itself. 

## Related issue

- [ticket](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/70682)

## Testing done

- Currently possible to input a non-numeric value in Day field resulting in a value of XX
- Changed Day input field to dropdown select utilizing DateUI Schema and validations
- Created previously non-existent unit test to validate render, the presence of error messages, and submission failure without completing required fields

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |mobile before|mobile after|
| Desktop |desktop before|desktop after|

## What areas of the site does it impact?

Form 526 > Veteran Details > Reserve and National Guard Information > Obligation start and end dates

## Acceptance criteria

- [x] Update date field to use the select element
- [x] Ensure change does not introduce impossible dates such as Feb 31st

### Quality Assurance & Testing

- [x] I added a unit test
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Comment by zack** [comment]
Were you able to reproduce getting "XX" through validation into the store? I see in your screenshot you have XX entered before blur, but after blur it should be validated and cleared.

File: src/applications/disability-benefits/all-claims/pages/reservesNationalGuardService.jsx [hunk 1]
```diff
@@ -22,30 +19,10 @@ export const uiSchema = {
...
-          'ui:errorMessages': {
-            pattern: 'Please enter a valid date',
-            required: 'Please enter a date',
-          },
-        },
-      },
-      unitName: {
-        'ui:title': 'Unit name',
+        from: { ...dateUI('Obligation start date') },
+        to: { ...dateUI('Obligation end date') },
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/pages/reservesNationalGuardService.jsx:23]: You don't really need to use the spread operator here, but I don't think it should be a blocker

```suggestion
        from: dateUI('Obligation start date'),
        to: dateUI('Obligation end date'),
```

**Review by Mottie — APPROVED** [review]

**Comment by freeheeling** [comment]
> Were you able to reproduce getting "XX" through validation into the store? I see in your screenshot you have XX entered before blur, but after blur it should be validated and cleared.

@zack 
Screen Shot 2023-11-29 at 9 30 27 AM

**Comment by zack** [comment]
> > Were you able to reproduce getting "XX" through validation into the store? I see in your screenshot you have XX entered before blur, but after blur it should be validated and cleared.
> 
> @zack Screen Shot 2023-11-29 at 9 30 27 AM

Incredible. Can you please share repro steps? Thanks!

**Comment by freeheeling** [comment]
> > > Were you able to reproduce getting "XX" through validation into the store? I see in your screenshot you have XX entered before blur, but after blur it should be validated and cleared.
> > 
> > 
> > @zack Screen Shot 2023-11-29 at 9 30 27 AM
> 
> Incredible. Can you please share repro steps? Thanks!

After selecting Month and entering valid Year, input **any** 2 characters (other than spaces) into the Day field, then click Continue.

**Comment by zack** [comment]
> > > > Were you able to reproduce getting "XX" through validation into the store? I see in your screenshot you have XX entered before blur, but after blur it should be validated and cleared.
> > > 
> > > 
> > > @zack Screen Shot 2023-11-29 at 9 30 27 AM
> > 
> > 
> > Incredible. Can you please share repro steps? Thanks!
> 
> After selecting Month and entering valid Year, input **any** 2 characters (other than spaces) into the Day field, then click Continue.

Super interesting. It looks like it fails validation immediately, but then if you refocus and blur another component it .. lets it go? I am curious if this is intentional for the [memorable date component](https://design.va.gov/components/form/memorable-date). We should probably ask platform. We want to be using their web components here, so if there's something preventing us from doing that we need to escalate it.

**Comment by Mottie** [comment]
> After selecting Month and entering valid Year, input any 2 characters (other than spaces) into the Day field, then click Continue.

That sounds like a validation function issue.

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Review by christinec-fftc — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/pages/reservesNationalGuardService.jsx (hunk 1 above)
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/pages/reservesNationalGuardService.jsx]: Made the requested modification.

**Pull request merged**